frontend/src/pages/TracesExplorer/__test__/TracesExplorer.test.tsx

```diff
@@ -81,6 +126,157 @@ const compositeQuery: Query = {
 };
 
 describe('TracesExplorer - ', () => {
+	// Initial filter panel rendering
+	// Test the initial state like which filters section are opened, default state of duration slider, etc.
+	it('should render the Trace filter', async () => {
+		const { getByText, getByTestId } = render(<Filter setOpen={jest.fn()} />);
+
+		Object.values(AllTraceFilterKeyValue).forEach((filter) => {
+			expect(getByText(filter)).toBeInTheDocument();
+		});
+
+		// Check default state of duration slider
+		const minDuration = getByTestId('min-input') as HTMLInputElement;
+		const maxDuration = getByTestId('max-input') as HTMLInputElement;
+		expect(minDuration).toHaveValue(null);
+		expect(minDuration).toHaveProperty('placeholder', '0');
+		expect(maxDuration).toHaveValue(null);
+		expect(maxDuration).toHaveProperty('placeholder', '100000000');
+
+		// Check which all filter section are opened by default
+		defaultOpenSections.forEach((section) =>
+			checkIfSectionIsOpen(getByTestId, section),
+		);
+
+		// Check which all filter section are closed by default
+		defaultClosedSections.forEach((section) =>
+			checkIfSectionIsNotOpen(getByTestId, section),
+		);
+
+		// check for the status section content
+		await checkForSectionContent(['Ok', 'Error']);
+
+		// check for the service name section content from API response
+		await checkForSectionContent([
+			'customer',
+			'demo-app',
+			'driver',
+			'frontend',
+			'mysql',
+			'redis',
+			'route',
+			'go-grpc-otel-server',
+			'test',
+		]);
+	});
+
+	// test the filter panel actions like opening and closing the sections, etc.
+	it('filter panel actions', async () => {
+		const { getByTestId } = render(<Filter setOpen={jest.fn()} />);
+
+		// Check if the section is closed
+		checkIfSectionIsNotOpen(getByTestId, 'name');
+		// Open the section
+		const name = getByTestId('collapse-name');
+		expect(name).toBeInTheDocument();
+
+		userEvent.click(within(name).getByText(AllTraceFilterKeyValue.name));
+		await waitFor(() => checkIfSectionIsOpen(getByTestId, 'name'));
+
+		await checkForSectionContent([
+			'HTTP GET',
+			'HTTP GET /customer',
+			'HTTP GET /dispatch',
+			'HTTP GET /route',
+		]);
+
+		// Close the section
+		userEvent.click(within(name).getByText(AllTraceFilterKeyValue.name));
+		await waitFor(() => checkIfSectionIsNotOpen(getByTestId, 'name'));
+	});
+
+	it('checking filters should update the query', async () => {
+		const { getByText } = render(
+			<QueryBuilderContext.Provider
+				value={
+					{
+						currentQuery: {
+							...initialQueriesMap.traces,
+							builder: {
+								...initialQueriesMap.traces.builder,
+								queryData: [initialQueryBuilderFormValues],
+							},
+						},
+						redirectWithQueryBuilderData,
+					} as any
+				}
+			>
+				<Filter setOpen={jest.fn()} />
+			</QueryBuilderContext.Provider>,
+		);
+
+		const okCheckbox = getByText('Ok');
+		fireEvent.click(okCheckbox);
```

Comment:
NIT: should we wrap the `fireEvent.click`... inside `act(() => )` (here, on line 243, and other places in the PR) :slightly_smiling_face:

Reply:
In most cases we dont need to wrap fireEvent or other statements with `act` as react does handle use flow itself, but we might still need to wrap statement with `waitFor`, `waitForElementToBeRemoved`, or `act` majorly for following cases - 
- Asynchronous Updates
- timers
- premature exits
- other update related thing

I will check again in my code if I missed some cases, Thanks for pointing out 🙌